jwst/datamodels/container.py

```diff
@@ -445,6 +467,77 @@ def ind_asn_type(self, asn_exptype):
                 ind.append(i)
         return ind
 
+    def set_buffer(self, buffer_size, overlap=None):
+        """Set buffer size for scrolling section-by-section access.
+
+        Parameters
+        ----------
+        buffer_size : float, None
+            Define size of buffer in Mb for each section.
+            If None, a default buffer size of 1Mb will be used.
+
+        overlap : int, optional
+            Define the number of rows of overlaps between sections.
+            If None, no overlap will be used.
+        """
+        buffMb = 1024 * 1024
+        self.buffer_size = buffMb if buffer_size is None else (buffer_size * buffMb)
+        self.overlap = 0 if overlap is None else overlap
+        self.grow = 0
+
+        with datamodel_open(self._models[0]) as model:
+            imrows, imcols = model.data.shape
+            data_item_size = model.data.itemsize
+            data_item_type = model.data.dtype
+            model.close()
+        del model
+
+        section_nrows = min(imrows, int(self.buffer_size / (imcols * data_item_size)))
+
+        if section_nrows == 0:
+            self.buffer_size = imcols * data_item_size
+            print("WARNING: Buffer size is too small to hold a single row.\n"
```

Comment:
Don't want print statements in production code. Change to log message.

Reply:
Looking at the code for this function, I think it needs serious revision. I'll do my best but @stsci-hack needs to review this. In principle he can fix the code later when working on `outlier_detection`. I am not using this part of the code.

Reply:
@hbushouse @stsci-hack done (fixed `set_buffer` and `print()`->logging).